{Release} Upgrade to Azure CLI 2.45.0 (#25342)
Azure/azure-cli: src/azure-cli/HISTORY.rst

**App Service**

* `az webapp create-remote-connection`: Update the host address of SSH tunnel from 0.0.0.0 to 127.0.0.1 (#25141)
* Add support to create ASPs with Memory Optimized Workers (#25198)
* Fix #17720: `az functionapp function`: Add new command to list functions in a function app (#24901)
* Fix #24285: `az webapp config access-restriction add`: Fix the bug that it does not support more than one Front Door ID in X-Azure-FDID (#25031)
* Fix #23603: `az functionapp config set`: Add new parameter to set PowerShell version (#25197)
* `az webapp config appsettings`: settings are registered as Deployment slot setting anytime when using `--slot-settings` (#24985)
* `az webapp config backup delete`: Add new command to delete a backup of the webapp (#25171)

**ARM**

* `az bicep`: Add configuration `bicep.use_binary_from_path`. Possible values include `if_running_in_ci` (default) and Booleans (#25033)
* `az bicep`: Add configuration `bicep.check_version` that accepts Boolean values. If set to `False`, version checks for Bicep CLI will be disabled (#25033)
* `az deployment what-if`: Fix an issue where formatting nested array changes throws an exception (#25152)

> * Fix #25022: `az resource tag`: Fix updating tag for Microsoft.insights/workbooks issue (#25142)

Reviewer: It's better to state that the "issue" is.

**ARO**

* `az aro create/update`: Add NetworkContributor role to NAT Gateways in Cluster Resource Group when creating or updating clusters (#25211)
* `az aro create`: Change `--pull-secret` parameter to no longer require `@` prefix on filenames (#25211)

**Backup**

* `az backup vault`: Add new parameter `--public-network-access` to support enabling public network access for the backup vault (#24699)
* `az backup vault create`: Add new parameter `--immutability-state` to support configuring immutability settings for the backup vault (#25277)

**Batch**

> * Fix: `az batch pool create`: Fix bug that caused error when parameter `--encryption-key-identifier` is used (#25034)

Reviewer 2: ```suggestion
* Fix #24007: `az batch pool create`: Fix bug that caused error when parameter `--encryption-key-identifier` is used (#25034)
```

> * Fix: `az batch pool create`: Fix bug that caused error when parameter `--encryption-key-identifier` is used (#25034)

Reviewer: Fix which issue?

> * Fix #24007: `az batch pool create`: Fix bug that caused error when parameter `--encryption-key-identifier` is used (#25034)

Reviewer 3: ```suggestion
* Fix #24007: `az batch pool create`: Fix bug that caused 'MissingRequiredProperty' error when parameter `--encryption-key-identifier` is used (#25034)
```

* `az image builder identity show`: Add this command to display managed identity of a image builder template (#25136)
* `az vmss reimage`: Let `--instance-id` support multiple ids (#25131)
* Fix #25308: `az disk create`: Fix help message for creating a standard disk for uploading blobs (#25315)
* `az vmss create/update`: Add new parameter `--enable-osimage-notification` to support enabling OS image scheduled event (#25273)
* `az vmss create`: Add new parameter `--max-surge` to support enabling rolling upgrade policy max surge (#25333)

**Cosmos DB**

* `az managed-cassandra datacenter update`: Add support to update `--sku` (#25215)

**DMS**

* `az dms project task create`: Update DMS MySQL API to support new migration types (#24978)

**Feedback**

* Stop including error messages in the feedback body (#25286)

**IoT**

* `az iot hub wait`: Add wait commands (#25326)
* `az iot hub delete`: Fix functionality issue for parameter `--no-wait` (#25326)

**Key Vault**

* `az keyvault security-domain restore-blob`: Support restoring blob offline (#25270)
* `az keyvault security-domain upload`: Add `--restore-blob` to prevent exposing keys in online environment (#25270)

**NetAppFiles**

> * Fix volume patch dataprotection props (#25231)

Reviewer 4: ```suggestion
* `az netappfiles volume update`: Fix volume patch dataprotection props (#25231)
```

**Network**

* [BREAKING CHANGE] `az network application-gateway ssl-profile`: Rename output property `verifyClientCertIssuerDn` to `verifyClientCertIssuerDN` to keep consistent with the name in API (#25143)
* [BREAKING CHANGE] `az network cross-region-lb frontended-ip`: Rename output properties with `publicIp` prefix to `publicIP` prefix to keep consistent with the names in API response (#25157)

> * [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `publicIp` prefix to `publicIP` prefix  to keep consistent with the names in API response (#25157)

Reviewer 5: ```suggestion
* [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `publicIp` prefix to `publicIP` prefix to keep consistent with the names in API response (#25157)
```

* [BREAKING CHANGE] `az network lb inbound-nat-rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25170)
* [BREAKING CHANGE] `az network lb rule`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network lb rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network cross-region-lb rule`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network cross-region-lb rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network lb outbound-rule`: Rename output property `frontendIpConfigurations` to `frontendIPConfigurations` to keep consistent with the name in API (#25281)

> * [BREAKING CHANGE] `az network cross-region-lb address-pool`:  Rename output property `loadBalancerFrontendIpConfiguration` to `loadBalancerFrontendIPConfiguration` to keep consistent with the name in API (#25317)
> * [BREAKING CHANGE] `az network cross-region-lb address-pool create`:  Replace preview argument `--backend-addresses-config-file --config-file` by `--backend-addresses` which supports Json, files and shorthand syntax formats (#25317)

Reviewer 6: ```suggestion
* [BREAKING CHANGE] `az network cross-region-lb address-pool`: Rename output property `loadBalancerFrontendIpConfiguration` to `loadBalancerFrontendIPConfiguration` to keep consistent with the name in API (#25317)
* [BREAKING CHANGE] `az network cross-region-lb address-pool create`: Replace preview argument `--backend-addresses-config-file --config-file` by `--backend-addresses` which supports Json, files and shorthand syntax formats (#25317)
```

> * `az bastion`: Move Azure Bastion to Azure CLI Extension (#25064)

Reviewer 6: ```suggestion
* [BREAKING CHANGE] `az bastion`: Move Azure Bastion to Azure CLI Extension (#25064)
```

> * [BREAKING CHANGE] `az bastion`: Move Azure Bastion to Azure CLI Extension `bastion` (#25064)

Reviewer 5: ```suggestion
* [BREAKING CHANGE] `az network bastion`: Move Azure Bastion to Azure CLI Extension `bastion` (#25064)
```

* Fix #25130: `az network list-usages`: `-o table` cannot be used (#25134)
* Fix #25124: `az network vnet-gateway create`: Active-Active gateway fails with insufficient IP configurations (#25156)
* `az network dns zone export`: Fix the export to emit all ALIAS records for a particular record set name (#25236)

> * `az network public-ip create`: Add parameter `--ddos-protection-plan` (#25264)

Reviewer 5: ```suggestion
* `az network public-ip create`: Add parameter `--ddos-protection-plan` to link a DDoS protection plan to public IP (#25264)
```

* Fix #25181: `az network nsg rule create`: Use `*` as default value for protocol (#25274)

> * `az network cross-region-lb address-pool update`: New command to update an address-pool (#25317)

Reviewer 6: ```suggestion
* `az network cross-region-lb address-pool update`: Add new command to update an address-pool (#25317)
```

> * `az network cross-region-lb address-pool update`:  New command to update an address-pool (#25317)

Reviewer 5: ```suggestion
* `az network cross-region-lb address-pool update`: New command to update an address-pool (#25317)
```

> **Packaging**
> 
> * Fix #24788: Add `-I` in entrypoint on MacOS (#24796)
> 
> **PolicyInsights**

Reviewer 4: ```suggestion
**PolicyInsights**
```

* `az policy attestation`: Add new command groups to manage resource policy attestation (#25230)

**RDBMS**

* `az mysql flexible-server create/update`: Add `--auto-scale-iops` to enable or disable autoscale of iops (#25153)

> * `az mysql flexible-server start/stop`: Add No Wait Support (#25212)
> * `az postgres flexible-server start/stop`: Add No Wait Support (#25212)

Reviewer 3: ```suggestion
* `az mysql flexible-server start/stop`: Add no-wait support (#25212)
* `az postgres flexible-server start/stop`: Add no-wait support (#25212)
```

* `az acr check-name`: Make command work with different profile (#25293)

**AKS**

* Make ContainerInsights DataCollectionRuleName consistent with Portal and other onboarding clients (#25050)
* `az aks upgrade`: Show warning if the Kubernetes version isn't supplied (#24800)
* `az aks create`: Deprecate parameters `--aad-client-app-id`, `--aad-server-app-id` and `--aad-server-app-secret` (#24999)
* `az aks update-credentials`: Deprecate parameters `--reset-aad`, `--aad-client-app-id`, `--aad-server-app-id` and `--aad-server-app-secret` (#24999)

**App Service**

* `az webapp create-remote-connection`: Update the host address of SSH tunnel from 0.0.0.0 to 127.0.0.1 (#25141)
* Add support to create ASPs with Memory Optimized Workers (#25198)
* Fix #17720: `az functionapp function`: Add new command to list functions in a function app (#24901)
* Fix #24285: `az webapp config access-restriction add`: Fix the bug that it does not support more than one Front Door ID in X-Azure-FDID (#25031)
* Fix #23603: `az functionapp config set`: Add new parameter to set PowerShell version (#25197)

> * `az webapp config appsettings`: settings are registered as Deployment slot setting anytime when using `--slot-settings` (#24985)

Reviewer 3: ```suggestion
* `az webapp config appsettings`: Register settings as Deployment slot setting anytime when using `--slot-settings` (#24985)
```

* `az webapp config backup delete`: Add new command to delete a backup of the webapp (#25171)

**ARM**

* `az bicep`: Add configuration `bicep.use_binary_from_path`. Possible values include `if_running_in_ci` (default) and Booleans (#25033)
* `az bicep`: Add configuration `bicep.check_version` that accepts Boolean values. If set to `False`, version checks for Bicep CLI will be disabled (#25033)
* `az deployment what-if`: Fix an issue where formatting nested array changes throws an exception (#25152)

> * Fix #25022: `az resource tag`: Fix updating tag for Microsoft.insights/workbooks issue (#25142)

Reviewer 3: ```suggestion
* Fix #25022: `az resource tag`: Fix the issue of `The serializedData field is missing or null` of updating tag for Microsoft.insights/workbooks (#25142)
```

* Fix #25308: `az disk create`: Fix help message for creating a standard disk for uploading blobs (#25315)
* `az vmss create/update`: Add new parameter `--enable-osimage-notification` to support enabling OS image scheduled event (#25273)
* `az vmss create`: Add new parameter `--max-surge` to support enabling rolling upgrade policy max surge (#25333)

**Cosmos DB**

* `az managed-cassandra datacenter update`: Add support to update `--sku` (#25215)

**DMS**

* `az dms project task create`: Update DMS MySQL API to support new migration types (#24978)
* Change Azure Database Migration Service string to Azure Database Migration Service (classic) (#25258)

**Feedback**

* Stop including error messages in the feedback body (#25286)

**IoT**

* `az iot hub wait`: Add wait commands (#25326)
* `az iot hub delete`: Fix functionality issue for parameter `--no-wait` (#25326)

**Key Vault**

* `az keyvault security-domain restore-blob`: Support restoring blob offline (#25270)
* `az keyvault security-domain upload`: Add `--restore-blob` to prevent exposing keys in online environment (#25270)

**NetAppFiles**

* Fix volume patch dataprotection props (#25231)

**Network**

> * [BREAKING CHANGE] `az network application-gateway ssl-profile`: Output property `verifyClientCertIssuerDn` is renamed by `verifyClientCertIssuerDN` to keep consistent with the name in API (#25143)
> * [BREAKING CHANGE] `az network cross-region-lb frontended-ip`: Rename output properties with `publicIp` prefix by `publicIP` prefix to keep consistent with the names in API response (#25157)
> * [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `publicIp` prefix by `publicIP` prefix  to keep consistent with the names in API response (#25157)
> * [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `privateIp` prefix by `privateIP` prefix to keep consistent with the names in API response (#25157)
> * [BREAKING CHANGE] `az network lb inbound-nat-pool`: Output property `enableFloatingIp` is renamed by `enableFloatingIP` to keep consistent with the name in API (#25170)
> * [BREAKING CHANGE] `az network lb inbound-nat-pool`: Output property `frontendIpConfiguration` is renamed by `frontendIPConfiguration` to keep consistent with the name in API (#25170)
> * [BREAKING CHANGE] `az network lb inbound-nat-rule`: Output property `enableFloatingIp` is renamed by `enableFloatingIP` to keep consistent with the name in API (#25170)
> * [BREAKING CHANGE] `az network lb inbound-nat-rule`: Output property `frontendIpConfiguration` is renamed by `frontendIPConfiguration` to keep consistent with the name in API (#25170)
> * [BREAKING CHANGE] `az network lb rule`: Output property `enableFloatingIp` is renamed by `enableFloatingIP` to keep consistent with the name in API (#25276)
> * [BREAKING CHANGE] `az network lb rule`: Output property `frontendIpConfiguration` is renamed by `frontendIPConfiguration` to keep consistent with the name in API (#25276)
> * [BREAKING CHANGE] `az network cross-region-lb rule`: Output property `enableFloatingIp` is renamed by `enableFloatingIP` to keep consistent with the name in API (#25276)
> * [BREAKING CHANGE] `az network cross-region-lb rule`: Output property `frontendIpConfiguration` is renamed by `frontendIPConfiguration` to keep consistent with the name in API (#25276)
> * [BREAKING CHANGE] `az network lb outbound-rule`: Output property `frontendIpConfigurations` is renamed by `frontendIPConfigurations` to keep consistent with the name in API (#25281)
> * [BREAKING CHANGE] `az network cross-region-lb address-pool`:  Output property `loadBalancerFrontendIpConfiguration` is renamed by `loadBalancerFrontendIPConfiguration` to keep consistent with the name in API (#25317)
> * [BREAKING CHANGE] `az network cross-region-lb address-pool create`:  Replace preview argument `--backend-addresses-config-file` and `--config-file` by `--backend-addresses` which supports Json, files and shorthand syntax formats (#25317)

Reviewer 6: ```suggestion
* [BREAKING CHANGE] `az network application-gateway ssl-profile`: Output property `verifyClientCertIssuerDn` is renamed to `verifyClientCertIssuerDN` to keep consistent with the name in API (#25143)
* [BREAKING CHANGE] `az network cross-region-lb frontended-ip`: Rename output properties with `publicIp` prefix to `publicIP` prefix to keep consistent with the names in API response (#25157)
* [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `publicIp` prefix to `publicIP` prefix  to keep consistent with the names in API response (#25157)
* [BREAKING CHANGE] `az network lb frontended-ip`: Rename output properties with `privateIp` prefix to `privateIP` prefix to keep consistent with the names in API response (#25157)
* [BREAKING CHANGE] `az network lb inbound-nat-pool`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25170)
* [BREAKING CHANGE] `az network lb inbound-nat-pool`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25170)
* [BREAKING CHANGE] `az network lb inbound-nat-rule`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25170)
* [BREAKING CHANGE] `az network lb inbound-nat-rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25170)
* [BREAKING CHANGE] `az network lb rule`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network lb rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network cross-region-lb rule`: Rename output property `enableFloatingIp` to `enableFloatingIP` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network cross-region-lb rule`: Rename output property `frontendIpConfiguration` to `frontendIPConfiguration` to keep consistent with the name in API (#25276)
* [BREAKING CHANGE] `az network lb outbound-rule`: Rename output property `frontendIpConfigurations` to `frontendIPConfigurations` to keep consistent with the name in API (#25281)
* [BREAKING CHANGE] `az network cross-region-lb address-pool`:  Rename output property `loadBalancerFrontendIpConfiguration` to `loadBalancerFrontendIPConfiguration` to keep consistent with the name in API (#25317)
* [BREAKING CHANGE] `az network cross-region-lb address-pool create`:  Replace preview argument `--backend-addresses-config-file --config-file` by `--backend-addresses` which supports Json, files and shorthand syntax formats (#25317)
```

Release History
===============

2.45.0
++++++

**ACR**

* [BREAKING CHANGE] `acr manifest list-referrers`: Support OCI reference types and remove ORAS artifact reference types (#25069)
* `az acr check-name`: Make command work with different profile (#25293)

**AKS**

* Make ContainerInsights DataCollectionRuleName consistent with Portal and other onboarding clients (#25050)
* `az aks upgrade`: Show warning if the Kubernetes version isn't supplied (#24800)
* `az aks create`: Deprecate parameters `--aad-client-app-id`, `--aad-server-app-id` and `--aad-server-app-secret` (#24999)
* `az aks update-credentials`: Deprecate parameters `--reset-aad`, `--aad-client-app-id`, `--aad-server-app-id` and `--aad-server-app-secret` (#24999)

**App Service**

* `az webapp create-remote-connection`: Update the host address of SSH tunnel from 0.0.0.0 to 127.0.0.1 (#25141)
* Add support to create ASPs with Memory Optimized Workers (#25198)

> * Fix#17720: `az functionapp function`: Add new command to list functions in a function app (#24901)

Reviewer 2: ```suggestion
* Fix #17720: `az functionapp function`: Add new command to list functions in a function app (#24901)
```

> * Fix#17720: `az functionapp function`: Add new command to list functions in a function app (#24901)

Reviewer 3: ```suggestion
* Fix #17720: `az functionapp function`: Add new command to list functions in a function app (#24901)
```

* Fix #24285: `az webapp config access-restriction add`: Fix the bug that it does not support more than one Front Door ID in X-Azure-FDID (#25031)
* Fix #23603: `az functionapp config set`: Add new parameter to set PowerShell version (#25197)

> * `az webapp config appsettings` settings are registered as Deployment slot setting anytime when using `--slot-settings` (#24985)

Reviewer 3: ```suggestion
* `az webapp config appsettings`: settings are registered as Deployment slot setting anytime when using `--slot-settings` (#24985)
```

**ARM**

* `az bicep`: Add configuration `bicep.use_binary_from_path`. Possible values include `if_running_in_ci` (default) and Booleans (#25033)
* `az bicep`: Add configuration `bicep.check_version` that accepts Boolean values. If set to `False`, version checks for Bicep CLI will be disabled (#25033)
* `az deployment what-if`: Fix an issue where formatting nested array changes throws an exception (#25152)
* Fix #25022: `az resource tag`: Fix updating tag for Microsoft.insights/workbooks issue (#25142)

**ARO**

* `az aro create/update`: Add NetworkContributor role to NAT Gateways in Cluster Resource Group when creating or updating clusters (#25211)
* `az aro create`: Change `--pull-secret` parameter to no longer require `@` prefix on filenames (#25211)

**Backup**

* `az backup vault`: Add new parameter `--public-network-access` to support enabling public network access for the backup vault (#24699)
* `az backup vault create`: Add new parameter `--immutability-state` to support configuring immutability settings for the backup vault (#25277)

**Batch**

* Fix: `az batch pool create`: Fix bug that caused error when parameter `--encryption-key-identifier` is used (#25034)

> * Test recording cleanup (#25166)

Reviewer 4: ```suggestion
```

> **Code**

Reviewer: Should be "Core"?

> 
> **Code**
> 
> * `aaz`: Support `AAZListArg` singular options list (#25099)

Reviewer 6: ```suggestion
```

**Compute**

* `az image builder identity assign`: Add this command to add managed identity to an existing image builder template (#25136)
* `az image builder identity remove`: Add this command to remove managed identity from an existing image builder template (#25136)

> * `az image builder identity show`: Add this command to display managed identity of a image builder template (#25136)

Reviewer 7: ```suggestion
* `az image builder identity show`: Add this command to display managed identity of an existing image builder template (#25136)
```

* `az vmss reimage`: Let `--instance-id` support multiple ids (#25131)
* Fix #25308: `az disk create`: Fix help message for creating a standard disk for uploading blobs (#25315)
* `az vmss create/update`: Add new parameter `--enable-osimage-notification` to support enabling OS image scheduled event (#25273)
* `az vmss create`: Add new parameter `--max-surge` to support enabling rolling upgrade policy max surge (#25333)

**Cosmos DB**

* `az managed-cassandra datacenter update`: Add support to update `--sku` (#25215)

**DMS**

* `az dms project task create`: Update DMS MySQL API to support new migration types (#24978)

> * Change Azure Database Migration Service string to Azure Database Migration Service (classic) (#25258)

Reviewer 2: ```suggestion
```

**Feedback**

* Stop including error messages in the feedback body (#25286)

**IoT**

* `az iot hub wait`: Add wait commands (#25326)
* `az iot hub delete`: Fix functionality issue for parameter `--no-wait` (#25326)

**Key Vault**

* `az keyvault security-domain restore-blob`: Support restoring blob offline (#25270)
* `az keyvault security-domain upload`: Add `--restore-blob` to prevent exposing keys in online environment (#25270)

**NetAppFiles**

* Fix volume patch dataprotection props (#25231)

**Network**

* [BREAKING CHANGE] `az network application-gateway ssl-profile`: Output property `verifyClientCertIssuerDn` is renamed by `verifyClientCertIssuerDN` to keep consistent with the name in API (#25143)

> * [BREAKING CHANGE] `az network cross-region-lb frontended-ip`: Rename output properties with `publicIp` prefix by `publicIP` prefix to keep consistent with the names in API response (#25157)

Reviewer: "Rename" should be followed by `to`, not `by`: https://english.stackexchange.com/questions/100296/which-preposition-to-use-with-rename
